packages/andromeda-finance/src/weighted_splitter.rs

```diff
@@ -0,0 +1,128 @@
+use common::{
+    ado_base::{modules::Module, recipient::Recipient, AndromedaMsg, AndromedaQuery},
+    error::ContractError,
+    require,
+};
+use cosmwasm_std::Uint128;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct AddressWeight {
+    pub recipient: Recipient,
+    pub weight: Uint128,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+/// A config struct for a `Splitter` contract.
+pub struct Splitter {
+    /// The vector of recipients for the contract. Anytime a `Send` execute message is sent the amount sent will be divided amongst these recipients depending on their assigned weight.
+    pub recipients: Vec<AddressWeight>,
+    /// Whether or not the contract is currently locked. This restricts updating any config related fields.
+    pub locked: bool,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct InstantiateMsg {
+    /// The vector of recipients for the contract. Anytime a `Send` execute message is
+    /// sent the amount sent will be divided amongst these recipients depending on their assigned weight.
+    pub recipients: Vec<AddressWeight>,
+    pub modules: Option<Vec<Module>>,
+}
+
+impl InstantiateMsg {
+    pub fn validate(&self) -> Result<bool, ContractError> {
```

Comment:
Since `true` is always returned when there is no error, we can replace `bool` with `()` instead.